File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/HttpBody.java
package com.roxiemobile.networkingapi.network.rest;

import com.roxiemobile.networkingapi.network.http.MediaType;

import org.jetbrains.annotations.NotNull;

public interface HttpBody {

    /**
     * TODO
     */
    @NotNull MediaType mediaType();

    /**
     * TODO
     */
    @NotNull byte[] body();
}

File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/HttpBody.kt
package com.roxiemobile.networkingapi.network.rest

import com.roxiemobile.networkingapi.network.http.MediaType

interface HttpBody {

// MARK: - Properties

    /**
     * TODO
     */
    val mediaType: MediaType

    /**
     * TODO
     */
    val body: ByteArray
}
